### PR TITLE
Initial implementation of 1Password ssh-agent

### PR DESCRIPTION
**Pull request by jfmcdowell**
# License Acceptance

- [X ] This repository is Apache version 2.0 licensed (some scripts may have alternate licensing inline in their code) and by making this PR, I am contributing my changes to the repository under the terms of the Apache 2 license.

## Description

This is an intital impleemntation of the 1Password ssh-agent to start the conversation on how to implement this correctly.

This code leverages the existing `can_haz` function to check for the 1Password CLI tool `op`. This is a **HUGE** assumption -- so lets talk about it.

However, I wonder if this code even needs to be merged because the user can set the [`SSH_AUTH_SOCK` environment variable](https://developer.1password.com/docs/ssh/get-started#step-4-configure-your-ssh-or-git-client) themselves in their functions file.

## Type of changes

- [X ] A helper script
- [ ] A link to an external resource like a blog post or video
- [ ] Text cleanups/updates
- [ ] Test updates
- [ ] Bug fix
- [ X] New feature
- [ ] Plugin list change

## Checklist

- [X ] I have read the **CONTRIBUTING** document.
- [ N/A] I have updated the readme if this PR changes/updates quickstart functionality.
- [ ] All new and existing tests pass.
- [ ] Any scripts added use `#!/usr/bin/env interpreter` instead of potentially platform-specific direct paths (`#!/bin/sh` is an allowed exception)
- [ ] Scripts are marked executable
- [ ] Scripts _do not_ have a language file extension unless they are meant to be sourced and not run standalone. No one should have to know if a script was written in bash, python, ruby or whatever. Not including file extensions makes it easier to rewrite the script in another language later without having to change every reference to the previous version.
- [ ] I have added a credit line to README.md for the script
- [ ] If there was no author credit in a script added in this PR, I have added one.
- [ ] I have confirmed that the link(s) in my PR are valid.

### CODE REVIEW

**Review by unixorn — CHANGES_REQUESTED** [review]
GH says there are merge conflicts. Once those are resolved this looks good.

File: zsh/.zshrc [hunk 1]
```diff
@@ -102,14 +102,17 @@ export LSCOLORS='Exfxcxdxbxegedabagacad'
...
 load-our-ssh-keys() {
-  if [ -z "$SSH_AUTH_SOCK" ]; then
-   # Check for a currently running instance of the agent
-   RUNNING_AGENT="$(ps -ax | grep 'ssh-agent -s' | grep -v grep | wc -l | tr -d '[:space:]')"
-   if [ "$RUNNING_AGENT" = "0" ]; then
-        # Launch a new instance of the agent
-        ssh-agent -s &> ~/.ssh/ssh-agent
-   fi
-   eval $(cat ~/.ssh/ssh-agent)
+  if can_haz op; then export SSH_AUTH_SOCK=~/Library/Group\ Containers/2BUA8C4S2C.com.1password/t/agent.sock
```
Comment (unixorn) [inline zsh/.zshrc:105]: `/Library/Group\ Containers/` does't exist on my Ventura machine. I'm using `op` and have my `SSH_AUTH_SOCK` set to `/private/tmp/com.apple.launchd.VYdKrAMX96/Listeners`. Did you have to create it while installing `op`?

**Review by unixorn — CHANGES_REQUESTED** [review]

File: zsh/.zshrc (hunk 1 above)
Comment (jfmcdowell) [reply zsh/.zshrc:105]: Hi Joe, this folder should come with the install of 1Password 8. Here is more info [from the docs.](https://developer.1password.com/docs/ssh/get-started)

Comment (unixorn) [reply zsh/.zshrc:105]: I'll take a look at the docs again today and let you know what I think. If it turns out that it generates a different ugly prefix to `com.1password` per machine it is probably best to go with adding a "After you install" section in the docs with a link to 1p's site.

Comment (jfmcdowell) [reply zsh/.zshrc:285]: I *think* everything is resolved and clean for your consideration.

**Review by unixorn — APPROVED** [review]
lgtm

**Comment by unixorn** [comment]
Sorry it took so long to get this merged

**Pull request merged**